contributing-device-support.md

```diff
@@ -1,120 +1,207 @@
+
+The root directory shall contain 2 directory entries:
+
+* a `layers` directory
+* [resin-yocto-scripts][resin-yocto-scripts] git submodule.
+
+_Note: you add submodules by `git submodule add <url> <directory>`, see the git documentation for more details._
+
+The root directory generally also includes the following files:
+
+* `CHANGELOG.md`
+* `LICENSE`
+* `README.md`
+* `VERSION`
+
+and one or more files named `<yocto-machine-name>.coffee`, one for each of the boards that the repository will add support for (eg. [`raspberry-pi3.coffee`](https://github.com/resin-os/resin-raspberrypi/blob/master/raspberrypi3.coffee) for Raspberry Pi 3 in `resin-raspberrypi`). This file contains information on the Yocto build for the specific board, in [CoffeeScript](http://coffeescript.org/) format. A minimal version of this file, using Raspberry Pi 3 as the example, would be:
+
+```
+module.exports =
+  yocto:
+    machine: 'raspberrypi3'
+    image: 'resin-image'
+    fstype: 'resin-sdcard'
+    version: 'yocto-jethro'
+    deployArtifact: 'resin-image-raspberrypi3.resin-sdcard'
+    compressed: true
+```
+
+The `layers` directory contains the git submodules of the yocto layers used in the build process. This normally means the following components are present:
+
+- [poky][poky]  at the version/revision required by the board BSP
 - [meta-openembedded][meta-openembedded] at the revision poky uses
 - [meta-resin][meta-resin] using the master branch
 - [oe-meta-go][oe-meta-go] using the master branch (there were no branches corresponding to the yocto releases at the time this howto was written)
-- yocto BSP layer for the board (for example, the BSP layer for Intel NUC is [meta-intel][meta-intel repo])
-- additional yocto layers required by the board BSP (check the yocto BSP layer of the respective board for instructions on how to build the BSP and what are the yocto dependencies of that particular BSP layer)
+- Yocto BSP layer for the board (for example, the BSP layer for Raspberry Pi is [meta-raspberrypi](https://github.com/agherzan/meta-raspberrypi))
+- any additional Yocto layers required by the board BSP (check the Yocto BSP layer of the respective board for instructions on how to build the BSP and what are the Yocto dependencies of that particular BSP layer)
+
+In addition to the above git submodules, the "layers" directory also contains a meta-resin-`<board-family>` directory (please note this directory is _not_ a git submodule, but an actual directory in the ). This directory contains the required customization for making a board resin.io enabled. For example, the [resin-raspberrypi](https://github.com/resin-os/resin-raspberrypi) repository contains the directory `layers/meta-resin-raspberrypi` to supplement the BSP from `layers/meta-raspberrypi` git submodule, with any changes that might be required by resinOS.
```

Comment:
"In addition to the above git submodules, the "layers" directory also contains a meta-resin-`<board-family>` directory " should be something like:

"In addition to the above git submodules, the "layers" directory also contains a meta-resin-`<board-name>` directory ", as usually there are many boards in a boards bsp. So for example, for the beaglebone board, the bsp layer is meta-ti and the directory we create is meta-resin-beaglebone